modules/tensor_mechanics/src/materials/crystal_plasticity/ComputeCrystalPlasticityVolumetricEigenstrain.C

```diff
@@ -0,0 +1,86 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ComputeCrystalPlasticityVolumetricEigenstrain.h"
+
+registerMooseObject("TensorMechanicsApp", ComputeCrystalPlasticityVolumetricEigenstrain);
+
+InputParameters
+ComputeCrystalPlasticityVolumetricEigenstrain::validParams()
+{
+  InputParameters params = ComputeCrystalPlasticityEigenstrainBase::validParams();
+  params.addClassDescription("Computes the deformation gradient from the volumetric eigenstrain "
+                             "due to spherical voids in a crystal plasticity simulation");
+  params.addRequiredParam<MaterialPropertyName>(
+      "spherical_void_number_density",
+      "The material property name of the number density of the spherical voids, in 1/mm^3.");
+  params.addRequiredParam<MaterialPropertyName>(
+      "mean_spherical_void_radius",
+      "The material property name for the mean radius value, in mm, for the spherical voids");
+
+  return params;
+}
+
+ComputeCrystalPlasticityVolumetricEigenstrain::ComputeCrystalPlasticityVolumetricEigenstrain(
+    const InputParameters & parameters)
+  : DerivativeMaterialInterface<ComputeCrystalPlasticityEigenstrainBase>(parameters),
+    _void_density(getMaterialProperty<Real>("spherical_void_number_density")),
+    _void_density_old(getMaterialPropertyOld<Real>("spherical_void_number_density")),
+    _void_radius(getMaterialProperty<Real>("mean_spherical_void_radius")),
+    _void_radius_old(getMaterialPropertyOld<Real>("mean_spherical_void_radius")),
+    _equivalent_linear_change(declareProperty<Real>("equivalent_linear_change")),
+    _sphere_coefficient(4.0 * (libMesh::pi) / 3.0)
```

Comment:
Minor comment: I would not create a member for this and if you do I suggest making it `static constexpr`. If you just use `4.0 * (libMesh::pi) / 3.0` below in the calculation the compiler will know to evaluate that at compile time, which it might not know in the present form of the PR.

Reply:
Thanks--change made